types/datum.go

```diff
@@ -1065,7 +1065,15 @@ func ProduceStrWithSpecifiedTp(s string, tp *FieldType, sc *stmtctx.StatementCon
 		// Flen is the rune length, not binary length, for Non-binary charset, we need to calculate the
```

Comment:
Flen is the length in:
* characters for nonbinary string types except *TEXT types.
* bytes for binary string types including *TEXT types.
(https://dev.mysql.com/doc/refman/5.7/en/storage-requirements.html#data-types-storage-reqs-strings)